### PR TITLE
[DOC, BUG] Clarify flag default value

### PR DESCRIPTION
**Pull request by eaasna**
seqan3 side of https://github.com/seqan/sharg-parser/issues/65

### CODE REVIEW

File: test/unit/argument_parser/argument_parser_design_error_test.cpp [hunk 1]
```diff
@@ -44,34 +44,41 @@ TEST(parse_test, design_error)
...
+    // default true
     seqan3::argument_parser parser4{"test_parser", 1, argv};
+    EXPECT_THROW(parser4.add_flag(true_value, 'i', "int", "oh oh default is true."),
+                 seqan3::design_error);
+
+    bool flag_value{false};
+
+    // short flag
+    seqan3::argument_parser parser5{"test_parser", 1, argv};
     parser4.add_flag(flag_value, 'i', "int1", "this is an int option.");
```
Comment (eseiler) [inline test/unit/argument_parser/argument_parser_design_error_test.cpp:58]: I think this line throws.

Change in line 27:

```cpp
int option_value;
```

to

```cpp
int option_value{};
```

If it's not initialised, its value is basically whatever was in that memory location before (it's almost certain not `0`). It then gets converted to `bool`. If `option_value` is anything but `0`, it will be converted to `true`.

**Review by eseiler — COMMENTED** [review]
Can you undo all the `{false}` changes? :D

File: test/unit/argument_parser/argument_parser_design_error_test.cpp (hunk 1 above)
Comment (eaasna) [reply test/unit/argument_parser/argument_parser_design_error_test.cpp:58]: But 
```c++
int option_value;  
```
is only used for the options and not flags. I don't think it would be converted to `bool`

Comment (eseiler) [reply test/unit/argument_parser/argument_parser_design_error_test.cpp:58]: Yes, I misread the variable name :D

File: doc/tutorial/argument_parser/index.md [hunk 2]
```diff
@@ -42,54 +42,55 @@ position instead of an identifier.
...
+as `_`, `-`, or `@`, but never start with `-`.
 * **Short identifiers**: must be unique and consist of only a single letter that is alphanumeric characters, `_` or `@`.
-* either the short or long id may be empty but not both at the same time.
+* Either the short or long id may be empty but not both at the same time.
 * Only the last positional option may be a list (see [lists](#section_list_positional_options)).
+* **Flags**: value must be false by default. Passing the flag identifier on the command line marks it as true.
 * The flag identifiers `-h`, `--help`, `--advanced-help`, `--advanced-help`, `--export-help`, `--version`, `--copyright`
-* are predefined and cannot be specified manually or used otherwise.
+are predefined and cannot be specified manually or used otherwise.
 * The seqan3::argument_parser::parse function may only be called once (per parser).
```
Comment (eaasna) [inline doc/tutorial/argument_parser/index.md:89]: Didn't see how to avoid the whitespace getting messed up. 

These lines are the actual `diff`

Comment (eseiler) [reply doc/tutorial/argument_parser/index.md:89]: Can you do a commit that only fixes the whitespaces?

See, e.g., here https://stackoverflow.com/questions/27489189/git-add-only-whitespace-changes

File: doc/tutorial/argument_parser/index.md [hunk 3]
```diff
@@ -411,16 +416,16 @@ Check if your options are set correctly by trying the following call:
...
+Our applications often do not allow just any value to be passed as input arguments and if we do not check for them,
+the program may exhibit undefined behaviour. The best way to carefully restrict user input is to directly check the
 input when parsing the command line. The seqan3::argument_parser provides **validators** for a given (positional) option.
 
-A *validator* is a [functor](https://stackoverflow.com/questions/356950/what-are-c-functors-and-their-uses) that is 
+A *validator* is a [functor](https://stackoverflow.com/questions/356950/what-are-c-functors-and-their-uses) that is
 called within the argument parser after retrieving and converting a command line argument. We provide several validators,
 which we hope cover most of the use cases, but you can always create your own validator
-(see section [Create your own validator](#section_create_your_own_validator)).
+(see section [Create your own validator](#cookbook_custom_validator)).
```
Comment (Irallia) [inline doc/tutorial/argument_parser/index.md:426]: Is this change intentional? Was the linking not correct before?

**Review by Irallia — CHANGES_REQUESTED** [review]
LGFM, just one question

File: doc/tutorial/argument_parser/index.md (hunk 3 above)
Comment (eaasna) [reply doc/tutorial/argument_parser/index.md]: My branch was behind the current master. The `(#cookbook_custom_validator)` version is correct and also in the current  master.

**Review by Irallia — APPROVED** [review]
